propolis/src/block/in_memory.rs

```diff
@@ -0,0 +1,279 @@
+use std::collections::VecDeque;
+use std::io::{Error, ErrorKind, Result};
+use std::num::NonZeroUsize;
+use std::sync::{Arc, Condvar, Mutex};
+
+use super::DeviceInfo;
+use crate::block;
+use crate::dispatch::{AsyncCtx, DispCtx, Dispatcher, SyncCtx, WakeFn};
+use crate::inventory::Entity;
+use crate::vmm::SubMapping;
+
+use tokio::sync::Semaphore;
+
+pub struct InMemoryBackend {
+    bytes: Arc<Mutex<Vec<u8>>>,
+
+    driver: Mutex<Option<Arc<Driver>>>,
+    worker_count: NonZeroUsize,
+
+    is_ro: bool,
+    block_size: usize,
+    sectors: usize,
+}
+
+impl InMemoryBackend {
+    pub fn create(bytes: Vec<u8>, is_ro: bool) -> Result<Arc<Self>> {
```

Comment:
Unless we're going to assert that the `bytes` length is a multiple of the sector size, is there anything here which demands a `Result`, rather than simply returning the created backend?

Reply:
checking that now in 6030809